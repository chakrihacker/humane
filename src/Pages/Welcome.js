import React from "react";
import styled from "styled-components";
import Button from '@material-ui/core/Button';
import { useHistory } from "react-router-dom";

const StyledWelcome = styled.div`
  ${({ theme }) => `
		background-color: ${theme.palette.primary.main};
		color: #fff;
		height: 100vh;
		display: flex;
		flex-direction: column;
		align-items: center;
		justify-content: center;
	`}
`;

const Row = styled.div`
	& > * {
		margin: 8px !important;
	}
	display: flex;
`

const StyledH1 = styled.h1`
	${({theme}) => `
    padding: ${theme.spacing(1)}px;
	`}
`

const Welcome = () => {
	let history = useHistory();

	const handleClick = (route) =>  {
		history.push(route)
	}
  return (
    <StyledWelcome>
      <StyledH1>Welcome to Humane</StyledH1>
			<Row>
				<Button variant="contained" color="secondary" onClick={() => handleClick("/login")}>Login</Button>
				<Button variant="contained" color="secondary">Sign up</Button>
			</Row>
    </StyledWelcome>
  );
};

export default Welcome;
