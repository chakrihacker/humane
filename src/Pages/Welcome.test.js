import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/react-testing";
import Welcome from "./Welcome";
import { ThemeProvider } from "styled-components";
import { createMuiTheme } from "@material-ui/core/styles";

const theme = createMuiTheme();

test("should match snapshot", () => {
  const {container} = render(
    <MockedProvider mocks={[]}>
      <ThemeProvider theme={theme}>
        <Welcome />
      </ThemeProvider>
    </MockedProvider>,
    { wrapper: MemoryRouter }
	);
	expect(container).toMatchSnapshot()
});
