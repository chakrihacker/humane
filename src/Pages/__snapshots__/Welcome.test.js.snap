// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm cKcCBR"
  >
    <h1
      class="sc-AxiKw civBZc"
    >
      Welcome to Humane
    </h1>
    <div
      class="sc-AxirZ fHLkvg"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Login
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Sign up
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
